checker/src/org/checkerframework/checker/index/samelen/SameLenAnnotatedTypeFactory.java

```diff
@@ -234,6 +242,44 @@ public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
         }
     }
 
+    @Override
+    public TreeAnnotator createTreeAnnotator() {
+        return new ListTreeAnnotator(
+                new SameLenTreeAnnotator(this),
+                new PropagationTreeAnnotator(this),
+                new ImplicitsTreeAnnotator(this));
+    }
+
+    protected class SameLenTreeAnnotator extends TreeAnnotator {
```

Comment:
Could you add documentation to indicate what this is intended to do?  The documentation might go here or elsewhere.